src/wechaty_puppet/memory_card/storage/backend_config.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import (
+    Optional,
+    Type,
+    Any,
+    Union,
+    TypeVar
+)
+
+
+@dataclass
+class StorageNopOptions:
+    placeholder: Optional[Any] = None
+
+
+StorageFileOptions = Type[StorageNopOptions]
+
+
+# @dataclass
+# class StorageS3Options:
+#     accessKeyId: str
+#     secretAccessKey: str
+#     region: str
+#     bucket: str
+
+
+@dataclass
+class StorageObsOptions:
+    accessKeyId: str
+    secretAccessKey: str
+    server: str
+    bucket: str
+    accessKeyId = ""
+    secretAccessKey = ""
+    server = ""
+    bucket = ""
+
+# TODO
+@dataclass
+class StorageBackendOptionsBase:
+    type: Optional[Any] = None
+
+
+@dataclass
+class StorageFileOptionsExtends(StorageBackendOptionsBase, StorageFileOptions):
+    type = "file"
+
+
+@dataclass
+class StorageNopOptionsExtends(StorageBackendOptionsBase, StorageNopOptions):
+    type = "nop"
+
+
+@dataclass
+class StorageObsOptionsExtends(StorageBackendOptionsBase, StorageObsOptions):
+    type = "obs"
+
+
+StorageBackendOptions = Union[StorageBackendOptionsBase, StorageFileOptionsExtends,
+                              StorageNopOptionsExtends, StorageObsOptionsExtends]
```

Comment:
In ts, it looks like this:


export type StorageBackendOptions =
    ({ type?: 'file' }  & StorageFileOptions)
  | ({ type?: 'nop' }   & StorageNopOptions)
  | ({ type?: 's3' }    & StorageS3Options)
  | ({ type?: 'obs' }   & StorageObsOptions)

How can I improve it ?

Reply:
Or maybe i am wrong?

Reply:
I think we should get into more experiment&test about differences between ts and python.